### PR TITLE
[Experimental] Add product filters edit link to product filters overlay inspector controls

### LINKED ISSUE #48641: Product Filters Redesign > Filters Overlay add edit link to product filters synced pattern
## Description

We need to add a link to the Filters Overlay block that points to the Product Filters Synced Pattern. Ideally just under the block description. But if this is not possible, we can add it to the inspector controls settings tab.

## Reference Image

Google Chrome -  Block  Product Filters – Figma 2024-06-19 at 1 57 06 PM

### PR DESCRIPTION
**Pull request by roykho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/48641

This PR adds the edit product filters template part link to the product filters overlay template part.

NOTE:

I did not find a way to add it below the block description. So I did the next best thing is to add it to the settings tab of the inspector control. But honestly I am not sure if adding this link adds any value? It might confuse the users more? Because you can already edit the template part in the canvas without having to go to the template part specifically and it will sync. Now that we give the edit link, users might ask "what's the difference between editing here vs going to the template part".

### Screenshot
Google Chrome - Filters Overlay (Experimental) ‹ Template part ‹ woo-blocks-dev ‹ Editor — WordPress 2024-07-30 at 5 47 30 AM

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Test by developers only**

**Prereq:**

Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.

1.  Go to the Site Editor->Patterns->General->Filters Overlay (Experimental).
2. Open the list view and click on the Product Filters Overlay (Experimental) block.
3. You should see in the inspector controls settings tab the "Edit product filters" link.
4. Click on it and ensure it takes you to the "Product filters" template part.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
[Experimental] Add edit product filters link to the product filters overlay template part
</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay/edit.tsx [hunk 1]
```diff
@@ -29,8 +32,23 @@ export const Edit = ( {
...
+
 	return (
 		<div { ...blockProps }>
+			<InspectorControls group="settings">
+				<PanelBody>
+					<ExternalLink
+						href={ templatePartEditUri }
+						className="wc-block-editor-product-filters__link"
+					>
+						{ __( 'Edit product filters', 'woocommerce' ) }
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay/edit.tsx]: The text here should be capitalized.

**Review by dinhtungdu — COMMENTED** [review]
Thanks for working on this PR.

I can confirm that I see the `Edit product filters` button and clicking on that button opens the Product Filters template part edit page on a new tab. Because we can't add the HTML to template part description, I think it's okay to use a setting panel for the link.

> But honestly I am not sure if adding this link adds any value? It might confuse the users more? Because you can already edit the template part in the canvas without having to go to the template part specifically and it will sync. Now that we give the edit link, users might ask "what's the difference between editing here vs going to the template part".

It's best to ask @jarekmorawski about this question. Jarek, to give you more context, in https://github.com/woocommerce/woocommerce/pull/49999, we embeded the `Product Filters` template part inside the `Filters Overlay` template part to keep the filers in sync.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay/edit.tsx (hunk 1 above)
Comment (roykho) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay/edit.tsx]: Good catch!

**Comment by roykho** [comment]
After speaking with Jarek, since there are already multiple options to edit the template part and the fact that the link cannot be place exactly where we wanted (under the block description), we decided not to implement this for now. So closing this.

**Pull request closed without merging**